### PR TITLE
chore: download nvidia drivers during VHD

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

This PR ensures that all nvidia drivers are downloaded during VHD creation, and installed from local files during CSE.

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [x] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: parts/k8s/cloud-init/artifacts/cse_helpers.sh [hunk 1]
```diff
@@ -17,14 +17,12 @@ else
   export GPU_DV=418.40.04
 fi
 export GPU_DEST=/usr/local/nvidia
-NVIDIA_DOCKER_VERSION=2.0.3
```
Comment (jackfrancis) [inline parts/k8s/cloud-init/artifacts/cse_helpers.sh:20]: none of these vars are re-used anywhere

**Comment by jackfrancis** [comment]
cc @Michael-Sinz @jadarsie @haofan-ms 

I haven't done a robust historical search, but I'm confident-ish that GPU node pools + airgap has never worked. This is a proposed solution to that.

**Comment by Michael-Sinz** [comment]
@jackfrancis Thank you for this work - we have not heavily invested in GPU work in air-gapped yet - we will be needing to do that soon so, thank you.

/lgtm

File: parts/k8s/cloud-init/artifacts/cse_helpers.sh [hunk 2]
```diff
@@ -17,14 +17,12 @@ else
...
-DOCKER_VERSION=1.13.1-1
-NVIDIA_CONTAINER_RUNTIME_VER='3.5.0*'
 PRIVATE_IP=$( (ip -br -4 addr show eth0 || ip -br -4 addr show azure0) | grep -Po '\d+\.\d+\.\d+\.\d+')
 if ! [[ $(echo -n "$PRIVATE_IP" | grep -c '^') == 1 ]]; then
   PRIVATE_IP=$(hostname -i)
 fi
 export PRIVATE_IP
+APT_CACHE_DIR=/var/cache/apt/archives/
```
Comment (Michael-Sinz) [inline parts/k8s/cloud-init/artifacts/cse_helpers.sh:25]: Not for now, but at some point we can look at cleaning up the cache after the node has fully provisioned.  (Just to free up space for containers/etc)

**Review by Michael-Sinz — APPROVED** [review]

File: parts/k8s/cloud-init/artifacts/cse_helpers.sh (hunk 2 above)
Comment (jackfrancis) [reply parts/k8s/cloud-init/artifacts/cse_helpers.sh:25]: Makes sense, and I think more generally the apt cache can probably be cleaned up after the bootstrap script finishes (probably in the background, non-blocking)

**Pull request merged**